### PR TITLE
[new] Add a training trick

### PR DESCRIPTION
**Pull request by LindaCY**
Description：
Add a training trick: Halve the learning rate if the performance on metrics not improving for [halve_lr_epochs] epochs, and then restart training by loading the previous best model.

"halve_lr_epochs" denotes the epochs of which  performance on metrics not improving. Default: -1 (never use it).

For example, we can use "halve_lr_epochs" as follows:
`
trainer = Trainer(model=model, n_epochs=100, optimizer=Adam(lr=0.01),validate_every=10,
                  train_data=train_data,
                  dev_data=dev_data,
                  loss=CrossEntropyLoss(),
                  metrics=AccuracyMetric(),use_tqdm=True, halve_lr_epochs=3
                  )
`

Main reason: Many empirical experiments have shown that this kind of trick can improve the efficiency of training and make the final performance of the model better.

Checklist  检查下面各项是否完成

Please feel free to remove inapplicable items for your PR.

-	[ ] The PR title starts with [$CATEGORY] (例如[bugfix]修复bug，[new]添加新功能，[test]修改测试，[rm]删除旧代码)
-	[ ] Changes are complete (i.e. I finished coding on this PR)  修改完成才提PR
-	[ ] All changes have test coverage  修改的部分顺利通过测试。对于fastnlp/fastnlp/*的修改，测试代码**必须**提供在fastnlp/test/*。
-	[ ] Code is well-documented  注释写好，API文档会从注释中抽取
-	[ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change  修改导致例子或tutorial有变化，请找核心开发人员

Changes: 逐项描述修改的内容
- 在trainer.py的Trainer类中添加了一个属性 halve_lr_epochs
- 在trainer.py中添加了一个训练trick相关代码，即如果在metric上的分数不再提升[halve_lr_epochs]个epoch，那么就减半学习率。然后从之前性能最好的模型开始重新训练。

Mention: 找人review你的PR

@修改过这个文件的人
@核心开发人员

### CODE REVIEW

**Comment by FengZiYjun** [comment]
Sorry for late reply. 
This trick could be implemented elegantly with the use of callbacks, without hard coding it into the trainer. I am considering merging this PR and move the changes to a new callback. 
Thanks for your contribution.

**Comment by LindaCY** [comment]
@FengZiYjun Thanks for your review~
